// jtopo/dist/link/link.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.default = undefined;

var _element = require('../element');

var _element2 = _interopRequireDefault(_element);

var _util = require('../util');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

let Link = class Link extends _element2.default {
	constructor(nodeA, nodeB, name) {
		super();
		this.nodeA = nodeA;
		this.nodeB = nodeB;
		this.name = name;
		this.style = { strokeStyle: '116, 166, 250', alpha: 1, lineWidth: 2 };
	}

	draw(ctx) {
		ctx.save();
		ctx.beginPath();
		ctx.strokeStyle = 'rgba(' + this.style.strokeStyle + ',' + this.style.alpha + ')';
		ctx.lineWidth = this.style.lineWidth;
		ctx.moveTo(this.nodeA.x + this.nodeA.width / 2, this.nodeA.y + this.nodeA.height / 2);
		ctx.lineTo(this.nodeB.x + this.nodeB.width / 2, this.nodeB.y + this.nodeB.height / 2);
		ctx.stroke();
		ctx.closePath();
		ctx.restore();
	}

	getLength() {
		return (0, _util.getDistance)(this.nodeA, this.nodeB);
	}
};
exports.default = Link;
