// jtopo/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _node = require('./node');

var _stage = require('./stage');

var _stage2 = _interopRequireDefault(_stage);

var _scene = require('./scene');

var _scene2 = _interopRequireDefault(_scene);

var _link = require('./link');

var _container = require('./container');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const JTopo = {
	constructor(name) {
		this.name = name;
	},
	Node: _node.Node,
	CircleNode: _node.CircleNode,
	EndPointNode: _node.EndPointNode,
	GhomboidNode: _node.GhomboidNode,
	HeartNode: _node.HeartNode,
	TextNode: _node.TextNode,
	TipNode: _node.TipNode,
	UMLClassNode: _node.UMLClassNode,
	Stage: _stage2.default,
	Scene: _scene2.default,
	Link: _link.Link,
	FoldLink: _link.FoldLink,
	CurverLink: _link.CurverLink,
	ArrowsLink: _link.ArrowsLink,
	ArrowsFoldLink: _link.ArrowsFoldLink,
	Container: _container.Container,
	GhomboidContainer: _container.GhomboidContainer,
	GridContainer: _container.GridContainer,
	OneItemContainer: _container.OneItemContainer,
	Version: '0.1.0',
	description: '参考与https://github.com/tuanjie54188/jtopo'
};

exports.default = JTopo;
